src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ActionPlanController.kt

```diff
@@ -42,7 +43,17 @@ class ActionPlanController(
 
     val draftActionPlanDTO = DraftActionPlanDTO.from(draftActionPlan)
     val location = locationMapper.mapToCurrentRequestBasePath("/{id}", draftActionPlanDTO.id)
-    return ResponseEntity.created(location).body(draftActionPlanDTO)
+    return ResponseEntity.created(location.toUri()).body(draftActionPlanDTO)
+  }
+
+  @PostMapping("/draft-action-plan/{id}/submit")
+  fun submitDraftActionPlan(
+    @PathVariable id: UUID,
+    authentication: JwtAuthenticationToken,
+  ): ActionPlanDTO {
+    val submittedByUser = jwtAuthUserMapper.map(authentication)
+    val submittedActionPlan = actionPlanService.submitDraftActionPlan(id, submittedByUser)
+    return ActionPlanDTO.from(submittedActionPlan)
```

Comment:
i think the front end contract expects a 201 response on success.

Reply:
but maybe that's a mistake on the contract.

Reply:
Yes it does and I have corrected it in the branch that includes the API changes agreed with George

`
val submittedActionPlanDTO = ActionPlanDTO.from(submittedActionPlan)
val location = locationMapper.mapToCurrentContextPathAsString("/action-plan/{id}", submittedActionPlanDTO.id)
return ResponseEntity.created(location.toUri()).body(actionPlanDTO)
`